clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -450,7 +451,21 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
     private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionResponse response, long correlationId,
                                long now) {
         Errors error = response.error;
-        if (error != Errors.NONE) {
+        if (error == Errors.MESSAGE_TOO_LARGE && batch.recordCount > 1) {
+            // If the batch is too large, we split the batch and send the split batches again. We do not decrement
+            // the retry attempts in this case.
+            log.warn("Got RecordBatchTooLargeException in correlation id {} on topic-partition {}, spitting and retrying ({} attempts left). Error: {}",
+                     correlationId,
+                     batch.topicPartition,
+                     this.retries - batch.attempts(),
+                     error);
```

Comment:
Isn't logging this redundant to including `RecordBatchTooLargeException` earlier in the message?